### PR TITLE
Manually patched exe support

### PR DESCRIPTION
**Pull request by TeapotWhiz**
Added support to manually patched game and editor, fixed LoadResourceFile.

### CODE REVIEW

File: H4API/lib/h4api/H4Base/H4Base.cpp [hunk 1]
```diff
@@ -96,12 +97,12 @@ namespace h4
 
 	_H4API_ BOOL8 H4Version::game() const
 	{
-		return m_version == GameVersion::GAME;
+		return m_version == GameVersion::GAME || m_version == GameVersion::UNPATCHED_GAME;
```
Comment (RoseKavalier) [inline H4API/lib/h4api/H4Base/H4Base.cpp]: I'd be inclined of doing a switcharoo here: define the original entry point as GAME/EDITOR and have the modified ones be written as LEGACY_X or something to that effect. Thoughts?

**Review by RoseKavalier — COMMENTED** [review]
Thanks, I never got around to fixing that address.

File: H4API/lib/h4api/H4Base/H4Base.cpp (hunk 1 above)
Comment (TeapotWhiz) [reply H4API/lib/h4api/H4Base/H4Base.cpp]: Makes sense if we assume that one standardized exe will be used to run mods.

Comment (RoseKavalier) [reply H4API/lib/h4api/H4Base/H4Base.cpp]: Agree, do you have time to make the modifications? If not I can do it this weekend.

Comment (TeapotWhiz) [reply H4API/lib/h4api/H4Base/H4Base.cpp]: Changed GAME to LEGACY_PATCHED_GAME, EDITOR to LEGACY_PATCHED_EDITOR, UNPATCHED_GAME and UNPATCHED_EDITOR just to GAME and EDITOR

**Review by RoseKavalier — APPROVED** [review]

File: H4API/lib/h4api/H4Base/H4Base.cpp (hunk 1 above)
Comment (RoseKavalier) [reply H4API/lib/h4api/H4Base/H4Base.cpp]: Great, many thanks!

**Pull request merged**